pallets/loans/src/functions.rs

```diff
@@ -253,7 +254,7 @@ impl<T: Config> Pallet<T> {
 				let new_borrowed_amount = loan_info
 					.borrowed_amount
 					.checked_add(&amount)
-					.ok_or(Error::<T>::AddAmountOverflow)?;
+					.ok_or(DispatchError::Arithmetic(ArithmeticError::Overflow))?;
```

Comment:
`DispatchError` implements `From<ArithmeticError>`. We should be able to just do `ok_or(ArithmeticError::Overflow)?` here and the `?` operator will handle the conversioe